framework/src/modules/chain/blocks/blocks.js

```diff
@@ -332,20 +327,18 @@ class Blocks extends EventEmitter {
 					await this._updateBroadhash();
 					this._lastBlock = newBlock;
 					this._isActive = false;
-					setImmediate(cb);
+					return;
 				} catch (error) {
 					this._isActive = false;
 					this.logger.error(error);
-					setImmediate(cb, error);
+					throw error;
```

Comment:
In case of the old implementation, the error in sequence is passed to the done parameter passed to the sequence.add, whereas now the error will bubble up in the call stack. Maybe we don't want that? I think we should replicate the functionality of sequences 100%